File: 프로그래머스/소수찾기.java
import java.util.*;
class Solution {
    static int answer = 0;
    static Set<Integer> set = new HashSet<>();
    static boolean[] visited = new boolean[7];
    public int solution(String numbers) {
        dfs(numbers,"");
        for(int i : set){
            if(isPrime(i)) answer++;
        }
        return answer;
    }
    static void dfs(String numbers, String temp){
        for(int i=0;i<numbers.length();i++){
            if(!visited[i]){
                visited[i] = true;
                set.add(Integer.parseInt(temp + numbers.charAt(i)));
                dfs(numbers, temp + numbers.charAt(i));
                visited[i] = false;
            }
        }
    }
    static boolean isPrime(int number){
        if(number < 2) return false;
        for(int i=2;i<=Math.sqrt(number);i++){
            if(number % i ==0) return false;
        }
        return true;
    }
}

File: 프로그래머스/소수찾기.js

function solution(numbers) {
    var answer = 0;
    const visited = new Array(numbers.length).fill(false);
    const filterNums = new Set();
    dfs(visited,filterNums,"",numbers);
    filterNums.forEach((v)=>{
        console.log(v);
        if(isPrime(v)) answer++;
    })
    
    return answer;
}

const dfs = (visited,filterNums,word,numbers) => {
    for(let i=0;i<numbers.length;i++){
        if(!visited[i]){
            filterNums.add(Number(word + numbers[i]));
            visited[i]=true;
            dfs(visited,filterNums,word + numbers[i],numbers);
            visited[i]=false;
        }
    }
}

const isPrime = (filterNum) => {
    if(filterNum < 2) return false;
    for(let i=2;i<=Math.sqrt(filterNum);i++){
        if(filterNum % i === 0) return false;
    }
    
    return true;
}
